### PR TITLE
Refactor AwsS3 implementation, move FeatureService to controllerBase

### PR DESCRIPTION
**Pull request by dexamundsen**
- AwsS3 open api schema: Refactor Generate Cloud name endpoint so that it can be reused in sagemaker
- AwsS3 controller implementation: refactor Credentials, GenerateCloudName, async delete t be reused in sagemaker
- Rename `Uuid resourceId` to `Uuid resourceUuid` & `Uuid workspaceId` to `Uuid workspaceUuid` 
- Move FeatureService from AwsController to ControllerBase, so that it can be used in all controllers. 
- Remove .swp file
- minor formatting changes
- change controllerBase.samService to protected field

No functional changes

### CODE REVIEW

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java [hunk 1]
```diff
@@ -208,14 +278,14 @@ public ResponseEntity<ApiCreatedControlledAwsS3StorageFolder> createAwsS3Storage
   @Traced
   @Override
   public ResponseEntity<ApiAwsS3StorageFolderResource> getAwsS3StorageFolder(
-      UUID workspaceUuid, UUID resourceId) {
+      UUID workspaceUuid, UUID resourceUuid) {
```
Comment (zloery) [inline service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java:281]: In most of the codebase I _think_ we use `workspaceId` and `resourceId` more than `_Uuid`. If the goal here is to standardize on one I'd prefer `_Id`, but I'm also fine using both versions.
Open to more input on that one, though

Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java:281]: This was ONLY to keep the naming consistent. 
in OpenApi `*id` is used
In implementation code `*uuid` was used more than `*id` (by overall count), hence this change.

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java [hunk 2]
```diff
@@ -113,23 +113,93 @@ private String getResourceRegion(Workspace workspace, String requestedRegion) {
...
+            workspaceUuid,
+            resourceUuid,
+            getAsyncResultEndpoint(jobControl.getId(), "delete-result"),
+            userRequest);
+
+    ApiDeleteControlledAwsResourceResult result = getAwsResourceDeleteResult(jobId);
+    return new ResponseEntity<>(result, getAsyncResponseCode(result.getJobReport()));
+  }
+
+  public ResponseEntity<ApiDeleteControlledAwsResourceResult> getDeleteAwsResourceResult(
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java:150]: Done

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java [hunk 3]
```diff
@@ -113,23 +113,93 @@ private String getResourceRegion(Workspace workspace, String requestedRegion) {
...
+        new ApiAwsCredential()
+            .version(1)
+            .accessKeyId(awsCredentials.accessKeyId())
+            .secretAccessKey(awsCredentials.secretAccessKey())
+            .sessionToken(awsCredentials.sessionToken())
+            .expiration(awsCredentials.expiration().atOffset(ZoneOffset.UTC)),
+        HttpStatus.OK);
+  }
+
+  public ResponseEntity<ApiAwsResourceCloudName> generateAwsS3StorageFolderCloudName(
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java:192]: added

File: openapi/src/parts/controlled_gcp_ai_notebook_instance.yaml [hunk 4]
```diff
@@ -62,11 +62,7 @@ paths:
         '500':
           $ref: '#/components/responses/ServerError'
     post:
-      summary: |
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply openapi/src/parts/controlled_gcp_ai_notebook_instance.yaml:65]: Intention was to keep it concise. the current message is not completely reflective of the reason - Openapi 3.x specifically disallows request body with DELETE operation. 
updated it instead.

**Review by zloery — APPROVED** [review]
LGTM pending workspaceId/workspaceUuid discussion above

**Review by yuhuyoyo — APPROVED** [review]
for the future, i'd rather review 10 prs that each does it own separate things than have a big one that does a bunch of the clean up.

**Comment by dexamundsen** [comment]
> for the future, i'd rather review 10 prs that each does it own separate things than have a big one that does a bunch of the clean up.

Sure

**Pull request merged**